### PR TITLE
New impl of GoogleLoadBalancerController

### PR DESCRIPTION
**Pull request by ttomsu**
Getting close to being able to disable `GoogleResourceRetriever`!

@duftler @lwander PTAL

tag: https://github.com/spinnaker/spinnaker/issues/712

### CODE REVIEW

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/controllers/GoogleLoadBalancerController2.groovy [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  static class GoogleLoadBalancerSummary {
+    String account
+    String region
+    String name
+    String type = GoogleCloudProvider.GCE
+  }
+
+  static class GoogleLoadBalancerDetails {
+    Long createdTime
+    String dnsname
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/controllers/GoogleLoadBalancerController2.groovy]: `dnsName`?

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/controllers/GoogleLoadBalancerController2.groovy [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    }
+
+    @JsonProperty("regions")
+    List<GoogleLoadBalancerAccountRegion> getRegions() {
+      mappedRegions.values() as List
+    }
+  }
+
+  static class GoogleLoadBalancerAccountRegion {
+    String name
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/controllers/GoogleLoadBalancerController2.groovy]: If this is the region, shouldn't we name it `region`?

Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/controllers/GoogleLoadBalancerController2.groovy]: eh, I guess I can live with it. Looked weird at first read, but I suppose it identifying the _name_ of the region is ok.

**Comment by duftler** [comment]
LGTM after addressing the one (potentially) relevant comment.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/controllers/GoogleLoadBalancerController2.groovy (hunk 1 above)
Comment (ttomsu) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/controllers/GoogleLoadBalancerController2.groovy]: Old impl has all lowercase.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/controllers/GoogleLoadBalancerController2.groovy (hunk 2 above)
Comment (ttomsu) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/controllers/GoogleLoadBalancerController2.groovy]: Agreed on it looking weird, but it matches the old impl's output verbatim.

**Pull request merged**